gcore/gdaldataset.cpp

```diff
@@ -1968,6 +1969,10 @@ CPLErr GDALDataset::IRasterIO( GDALRWFlag eRWFlag,
 
     CPLAssert(nullptr != pData);
 
+    RawDataset* rawDataset = dynamic_cast<RawDataset*>(this);
```

Comment:
not pretty to put that here. That should be put in RawDataset::IRasterIO() itself

Reply:
Ok, inheritance is more complex that I thought. I believed that RawDataset::IRasterIO was somwhat totally overriden